homeassistant/components/apcupsd/__init__.py

```diff
@@ -99,6 +90,22 @@ def statflag(self) -> str | None:
         """Return the STATFLAG indicating the status of the UPS, if available."""
         return self.status.get("STATFLAG")
 
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return the DeviceInfo of this APC UPS for the sensors, if serial number is available."""
+        if self.serial_no is None:
+            return None
+
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.serial_no)},
+            model=self.model,
+            manufacturer="APC",
+            name=self.name,
```

Comment:
The `name` property is also used in `config_flow.py` to be displayed as the title of the integration (if available).

Reply:
BTW, if the concern is that we shouldn't have `name` property to avoid confusions, I have another PR #88467 to replace this class with `DataUpdateCoordinator`, and renamed this property to `ups_name`.

(I'll rebase and further simplify that PR once this PR is merged)